op-chain-ops/foundry/allocs.go

```diff
@@ -0,0 +1,123 @@
+package foundry
+
+import (
+	"encoding/json"
+	"fmt"
+	"maps"
+	"math/big"
+	"os"
+
+	"github.com/holiman/uint256"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type ForgeAllocs struct {
+	Accounts types.GenesisAlloc
+}
+
+// FromState takes a geth StateDB, and dumps the accounts into the ForgeAllocs.
+// Any previous allocs contents are removed.
+// Warning: the state must be committed first, trie-key preimages must be present for iteration,
+// and a fresh state around the committed state-root must be presented, for the latest state-contents to be dumped.
+func (f *ForgeAllocs) FromState(stateDB StateDB) {
+	f.Accounts = make(types.GenesisAlloc)
+	stateDB.DumpToCollector((*forgeAllocsDump)(f), &state.DumpConfig{
+		OnlyWithAddresses: true,
+	})
+}
+
+// StateDB is a minimal interface to support dumping of Geth EVM state to ForgeAllocs.
+type StateDB interface {
+	DumpToCollector(c state.DumpCollector, conf *state.DumpConfig) (nextKey []byte)
+}
+
+// Assert that the Geth StateDB implements this interface still.
+var _ StateDB = (*state.StateDB)(nil)
+
+// forgeAllocsDump is a wrapper to hide the error-prone state-dumping interface from public API.
+// Use ForgeAllocs.FromState to dump a state to forge-allocs.
+type forgeAllocsDump ForgeAllocs
+
+// ForgeAllocs implements state.DumpAllocator, such that the EVM state can be dumped into it:
+// with a StateDB.DumpToCollector call.
+var _ state.DumpCollector = (*forgeAllocsDump)(nil)
+
+func (d *forgeAllocsDump) OnRoot(hash common.Hash) {
+	// Unlike the geth raw-state-dump, forge-allocs do not reference the state trie root.
+}
+
+func (d *forgeAllocsDump) OnAccount(address *common.Address, account state.DumpAccount) {
+	fmt.Printf("dumping %v %v\n", address, account)
```

Comment:
Did you mean to leave this print statement in?

Reply:
Whoops, no, thanks